File: CS/WindowsFormsApplication399/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication399 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            gridControl1.VisibleChanged += gridControl1_VisibleChanged;
            gridControl2.VisibleChanged += gridControl1_VisibleChanged;
        }

        void gridControl1_VisibleChanged(object sender, EventArgs e) {
            transitionManager1.EndTransition();
        }

        private void navBarControl1_LinkClicked(object sender, DevExpress.XtraNavBar.NavBarLinkEventArgs e) {
            switch (e.Link.Caption) {
                case "Grid1":
                    transitionManager1.StartTransition(panelControl1);
                    gridControl2.Visible = false;
                    gridControl1.Visible = true;
                    break;
                case "Grid2":
                    transitionManager1.StartTransition(panelControl1);
                    gridControl2.Visible = true;
                    gridControl1.Visible = false;
                    break;
                default:
                    break;
            }
        }
    }
}

File: VB/WindowsFormsApplication399/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace WindowsFormsApplication399
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			AddHandler gridControl1.VisibleChanged, AddressOf gridControl1_VisibleChanged
			AddHandler gridControl2.VisibleChanged, AddressOf gridControl1_VisibleChanged
		End Sub

		Private Sub gridControl1_VisibleChanged(ByVal sender As Object, ByVal e As EventArgs)
			transitionManager1.EndTransition()
		End Sub

		Private Sub navBarControl1_LinkClicked(ByVal sender As Object, ByVal e As DevExpress.XtraNavBar.NavBarLinkEventArgs) Handles navBarControl1.LinkClicked
			Select Case e.Link.Caption
				Case "Grid1"
					transitionManager1.StartTransition(panelControl1)
					gridControl2.Visible = False
					gridControl1.Visible = True
				Case "Grid2"
					transitionManager1.StartTransition(panelControl1)
					gridControl2.Visible = True
					gridControl1.Visible = False
				Case Else
			End Select
		End Sub
	End Class
End Namespace
